third-party/realdds/src/dds-metadata-syncer.cpp

```diff
@@ -0,0 +1,95 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include <realdds/dds-metadata-syncer.h>
+#include <realdds/dds-utilities.h>
+
+
+namespace realdds {
+
+
+void dds_metadata_syncer::enqueue_frame( key_type id, frame_holder && frame )
+{
+    std::lock_guard< std::mutex > lock( _queues_lock );
+    while( _frame_queue.size() >= max_frame_queue_size )
```

Comment:
I think that with your changes max_frame_queue_size should be 1.
If no metadata is received, we want to delay the frame for maximum of one other frame. Here it can be delayed for 2 frames.

Reply:
What changes are you referring to, that we can change the limit? I'm not objecting (though I'm hesitant to change this PR - I prefer waiting until after I have the unit-tests in), just curious.

Reply:
Changes like checking the queue size here and not in `search_for_match`. We can leave for now, until the unit-tests will be completed. Just need to make sure not to delay the frame for too long